### ISSUE TITLE
Grafana with Persistence Failing (EFS)

### ISSUE THREAD

**Issue by etotheipi** [helm/charts#1223]
Looks a lot like https://github.com/kubernetes/charts/issues/1071 but slightly different error.  In that case the grafana DB is created, but then subsequent errors.  In this case, grafana can't even create the DB.

I'm using stable/grafana 0.3.6, with persistent storage backed by EFS (using external provisioner from https://github.com/kubernetes-incubator/external-storage/tree/master/aws/efs).  

The PVs/PVCs from this EFS storage has been successfully used to run a persistent zookeeper statefulset and the influxdb helm chart.    The grafana helm chart **does** work when I disable persistence.

But when I try to enable persistence, I have tried both pre-creating a PVC and providing it as an `existingClaim` as well as just providing the `storageClass`.  In both cases I simply see `unable to open database file` at the end of the logs of the crashed grafana pod.

Here's the section in the values.yaml file:
```
  persistentVolume:
    enabled: true
    accessModes:
      - ReadWriteMany
    existingClaim: "grafana-claim"
    #size: 1Gi
    #storageClass: primary
```

This is the logs after the pod enters "Error" status:

```
$ k logs grafana-grafana-3005102734-dj8cj -nmon
t=2017-06-01T16:40:04+0000 lvl=info msg="Starting Grafana" logger=main version=4.3.2 commit=ed4d170 compiled=2017-05-31T08:48:16+0000
t=2017-06-01T16:40:04+0000 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2017-06-01T16:40:04+0000 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from command line" logger=settings arg="default.log.mode=console"
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_USER=admin"
t=2017-06-01T16:40:04+0000 lvl=info msg="Config overriden from Environment variable" logger=settings var="GF_SECURITY_ADMIN_PASSWORD=*********"
t=2017-06-01T16:40:04+0000 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2017-06-01T16:40:04+0000 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana/data
t=2017-06-01T16:40:04+0000 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2017-06-01T16:40:04+0000 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2017-06-01T16:40:04+0000 lvl=info msg="Initializing DB" logger=sqlstore dbtype=sqlite3
t=2017-06-01T16:40:04+0000 lvl=info msg="Starting DB migration" logger=migrator
t=2017-06-01T16:40:04+0000 lvl=eror msg="Fail to initialize orm engine" logger=sqlstore error="Sqlstore::Migration failed err: unable to open database file\n"
```

In the both cases, when I mount the NFS volume and look at the directories, the grafana dir is empty.   Also, grafana is modifying the owner/group of claim.  The claim started as `root:2004` and is modified by grafana to be `104:107` as seen below:

```
[root@pv-browse mnt]# ll                                                                                                                                                            
total 20
d---rws--x 2  104  107 6144 Jun  1 17:05 grafana-claim-pvc-7d13abe4-46ec-11e7-99e0-0e2a4fb0f77c
d---rws--x 5 root 2005 6144 Jun  1 16:55 influxdb-influxdb-pvc-14927d54-46eb-11e7-aec4-06f8921f8b0a
d---rws--x 3 root 2000 6144 Jun  1 15:45 zk-vol-zookeeper-0-pvc-649e31e4-46e1-11e7-99e0-0e2a4fb0f77c
...
```

Perhaps this is a permissions things.  But it seems that the grafana pod is capable of changing the permissions on the claim, so I don't know why it would be unable to write data to the same claim.  

The pod is not up long enough for me to `exec` into it and test writing to the PVC.

For completeness: 

```
 $ kubectl -n mon get po
NAME                                  READY     STATUS    RESTARTS   AGE
grafana-grafana-3005102734-j2052      0/1       Error     2          5m
influxdb-influxdb-1700725659-s0kcm    1/1       Running   0          16m
...
```

```$ kubectl -n mon get pv,pvc
NAME                                          CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS    CLAIM                        STORAGECLASS   REASON    AGE
pv/pvc-14927d54-46eb-11e7-aec4-06f8921f8b0a   10Gi       RWX           Delete          Bound     mon/influxdb-influxdb        primary                  16m
pv/pvc-7d13abe4-46ec-11e7-99e0-0e2a4fb0f77c   1Gi        RWX           Delete          Bound     mon/grafana-claim            primary                  6m
...

NAME                    STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS   AGE
pvc/grafana-claim       Bound     pvc-7d13abe4-46ec-11e7-99e0-0e2a4fb0f77c   1Gi        RWX           primary        6m
pvc/influxdb-influxdb   Bound     pvc-14927d54-46eb-11e7-aec4-06f8921f8b0a   10Gi       RWX           primary        16m
```

```
 $ kubectl -n mon describe po grafana-grafana-3005102734-j2052  
Name:		grafana-grafana-3005102734-j2052
Namespace:	mon
...
Containers:
  grafana:
    ...
    Mounts:
      /etc/grafana from config-volume (rw)
      /var/lib/grafana/dashboards from dashboard-volume (rw)
      /var/lib/grafana/data from storage-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-2cjms (ro)
...
Volumes:
  ...
  storage-volume:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	grafana-claim
    ReadOnly:	false
Events:
  FirstSeen	LastSeen	Count	From					SubObjectPath			Type		Reason		Message
  ---------	--------	-----	----					-------------			--------	------		-------
  7m		7m		1	default-scheduler							Normal		Scheduled	Successfully assigned grafana-grafana-3005102734-j2052 to ip-172-20-51-91.ec2.internal
  7m		7m		1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal		Started		Started container with id 3903486b8182c833054b5a0161de0117e7f3367d2ff1e807083c268b1ecaf32a
  7m		7m		1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal		Created		Created container with id 3903486b8182c833054b5a0161de0117e7f3367d2ff1e807083c268b1ecaf32a
  7m		7m		1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal		Created		Created container with id 0028c5205866264dd56f7891226e9f3974cbb0deac104a7470a42241979cea1d
  7m		7m		1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal		Started		Started container with id 0028c5205866264dd56f7891226e9f3974cbb0deac104a7470a42241979cea1d
  7m		7m		2	kubelet, ip-172-20-51-91.ec2.internal					Warning		FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "grafana" with CrashLoopBackOff: "Back-off 10s restarting failed container=grafana pod=grafana-grafana-3005102734-j2052_mon(972042a0-46ec-11e7-99e0-0e2a4fb0f77c)"

  7m	7m	3	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal	Pulled		Successfully pulled image "grafana/grafana:latest"
  7m	7m	3	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal	Pulling		pulling image "grafana/grafana:latest"
  7m	7m	1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal	Created		Created container with id 7656e3171d07bd65dc352dffe832941596fc5b3a5594281345bda574d3032f01
  7m	7m	1	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Normal	Started		Started container with id 7656e3171d07bd65dc352dffe832941596fc5b3a5594281345bda574d3032f01
  7m	7m	3	kubelet, ip-172-20-51-91.ec2.internal	spec.containers{grafana}	Warning	BackOff		Back-off restarting failed container
  7m	7m	1	kubelet, ip-172-20-51-91.ec2.internal					Warning	FailedSync	Error syncing pod, skipping: failed to "StartContainer" for "grafana" with CrashLoopBackOff: "Back-off 20s restarting failed container=grafana pod=grafana-grafana-3005102734-j2052_mon(972042a0-46ec-11e7-99e0-0e2a4fb0f77c)"

  6m	7s	32	kubelet, ip-172-20-51-91.ec2.internal		Warning	FailedSync	Error syncing pod, skipping: rpc error: code = 2 desc = Error: No such container: 0028c5205866264dd56f7891226e9f3974cbb0deac104a7470a42241979cea1d
```

**Comment by etotheipi** [comment]
So I had exec'd into the container and confirmed that I could touch files in the PVC mount (as root), but didn't realize that entrypoint script was run as root but de-escalates to user `grafana` at the end.  I confirmed that user `grafana` **cannot** write to the EFS volumes.  

The same thing happened for elasticsearch.  This may actually be a permissions issue related to NFS mounts, or EFS PVs in general.  Probably not for this repo.  Closing.

https://github.com/kubernetes-incubator/external-storage/issues/147
